manorMatch/src/components/UserProfile/UserProfile.jsx

```diff
@@ -0,0 +1,102 @@
+import NavBar from '../../utils/NavBar.jsx';
+import axios from 'axios';
+import { useState, useEffect } from 'react';
+import BookingHistory from './BookingHistory.jsx';
+
+const UserProfile = () => {
+
+  const [currentUser, setCurrentUser] = useState({});
+  const [bookings, setBookings] = useState([]);
+  const [previousBookings, setPreviousBookings] = useState([]);
+  const [upcomingBookings, setUpcomingBookings] = useState([]);
+  const [activeTab, setActiveTab] = useState('upcoming');
+
+  const getBookings = (id) => {
+    const apiUrl = import.meta.env.VITE_API_URL;
+    axios.get(`${apiUrl}/bookings`, {
+      params: {
+        userId: id
+    }})
+    .then((response) => {
+      const bookings = response.data
+
+      const vendorPromises = bookings.map((booking) => {
+        return booking.services.map((service) => {
+          return axios.get(`${apiUrl}/vendorInfo`, {
+            params: {
+              id: service.vendorId
+            }
+          })
+          .then((vendorResponse) => {
+            service.vendorPhoto = vendorResponse.data.photo;
+            service.vendorName = vendorResponse.data.name;
+          })
+        })
+      })
+
+      return Promise.all(vendorPromises.flat()).then(() => bookings);
+    })
+    .then((bookings) => {
+      const upcomingBookings = bookings.filter((booking) => new Date(booking.jobDate) > new Date());
+      const previousBookings = bookings.filter((booking) => new Date(booking.jobDate) < new Date());
+
+      setBookings(bookings);
+      setPreviousBookings(previousBookings);
+      setUpcomingBookings(upcomingBookings);
+    })
+
+    .catch((error) => {
+      console.error('Error:', error);
+    })
+}
+
+  useEffect(() => {
+    const getUserAndBookings = () => {
+      const apiUrl = import.meta.env.VITE_API_URL;
+      const userEmail = 'Naomie_Fadel65@yahoo.com';
```

Comment:
Testing!  Here!

Reply:
Hard coding: out! Getting from local storage: in!